### PR TITLE
frontend: edit docs for playwright

### PR DESCRIPTION
**Pull request by vyncent-t**
# Update Documentation for Local Playwright End-to-End Tests

## Description

This PR updates the existing documentation to include more detailed instructions on setting up and running local Playwright end-to-end tests. The enhancements aim to provide developers with clear guidelines on how to effectively utilize Playwright for testing their local developments, ensuring that they can quickly and efficiently verify changes before deployment.

## Changes

- [x] Expanded the documentation to include step-by-step setup instructions for running Playwright locally.
- [X] Added options for preferences using configurations to playwright.config.ts

## Verification

- [x] Ensured that all new documentation content is clear, accurate, and easy to follow.
- [x] Tested the setup instructions and test execution steps to ensure they are applicable and functional.
- [x] Reviewed the documentation for technical accuracy and coherence with existing documentation standards.

### CODE REVIEW

**Comment by sniok** [comment]
Something that could be useful is a development guide, like how to create, change and validate tests and also how to run tests against a local frontend build. Also we should probably mention the `playwright --ui`[0] option which I think is really useful

[0] https://playwright.dev/docs/test-ui-mode

File: e2e-tests/README.md [hunk 1]
```diff
@@ -4,23 +4,42 @@
 npx playwright install
 ```
 
-The instructions below assume Headlamp is running in-cluster with minikube.
+The instructions below assume Headlamp is running locally in-cluster with minikube.
```
Comment (joaquimrocha) [inline e2e-tests/README.md:7]: I think this is implicit and less confusing to leave as it was.

**Review by joaquimrocha — COMMENTED** [review]
Left a comment. Otherwise, it feels like Olek's idea of mentioning --ui is a good one.

File: e2e-tests/README.md (hunk 1 above)
Comment (vyncent-t) [reply e2e-tests/README.md:7]: done!

**Comment by joaquimrocha** [comment]
@skoeva Can you confirm/test these steps?

**Review by skoeva — DISMISSED** [review]
other than these changes, LGTM

File: e2e-tests/README.md [hunk 2]
```diff
@@ -38,3 +62,17 @@ npx playwright test -g "404 page is present"
...
 npx playwright test -g "404 page is present" --headed
 ```
+
+## Optional configuration
+
+### Headed vs Headless
+
+- If you wish to see the test run in a real browser, you can add the `--headed` flag to the command.
+
+- You can also modify the playwright.config.js file to change the browser that is used for the tests. You must be sure not to add this change to the repository when you push.
```
Comment (skoeva) [inline e2e-tests/README.md:72]:
```suggestion
- You can also modify the playwright.config.ts file to change the browser that is used for the tests. You must be sure not to add this change to the repository when you push.
```

File: e2e-tests/README.md [hunk 3]
```diff
@@ -38,3 +62,17 @@ npx playwright test -g "404 page is present"
...
+  - within the playwright.config.js file locate the `const config: PlaywrightTestConfig` object. within the object, modify the use object to contain a field for headless set to false ex. `use: { ..., headless: false }` property to run the tests in a real browser.
+
+### Slow down the tests
+
+- You can modify the playwright.config.js file to slow down the tests. You must be sure not to add this change to the repository when you push.
```
Comment (skoeva) [inline e2e-tests/README.md:77]:
```suggestion
- You can modify the playwright.config.ts file to slow down the tests. You must be sure not to add this change to the repository when you push.
```

**Comment by vyncent-t** [comment]
rebased and added a section for the UI mode

**Review by illume — APPROVED** [review]
🎉🎈 Thanks.

**Pull request merged**